controllers/grid/queries/QueriesGridHandler.inc.php

```diff
@@ -320,6 +320,22 @@ function deleteQuery($args, $request) {
 		$notificationDao = DAORegistry::getDAO('NotificationDAO');
 		$notificationDao->deleteByAssoc(ASSOC_TYPE_QUERY, $query->getId());
 
+		// Update submission notifications
+		$notificationsToUpdate = array(
```

Comment:
If the notification manager delegates are aligned into one class (once the notification stuff is removed), maybe it'll be possible to remove the stage-specific stuff from here? The query grid classes so far don't need to behave differently based on what stage they're in, and that would be nice to preserve.